src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/DecompressionHandler.cs

```diff
@@ -122,6 +122,14 @@ public DecompressedContent(HttpContent originalContent)
 
             protected abstract Stream GetDecompressedStream(Stream originalStream);
 
+            protected override void SerializeToStream(Stream stream, TransportContext? context, CancellationToken cancellationToken)
+            {
+                using (Stream decompressedStream = CreateContentReadStream(cancellationToken))
+                {
+                    decompressedStream.CopyTo(stream);
+                }
+            }
```

Comment:
If the base SerializeToStream throws a NotSupportedException, does this mean we have a test hole?  Otherwise, I'd have expected the lack of an override here to result in NSEs in the HttpClient.Send PR.

Reply:
We have a test hole, [Decompression tests ](https://github.com/dotnet/runtime/blob/master/src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.Decompression.cs) aren't being run for sync scenario. I'm adding them.
It manifest if you have buffered, decompressed content.

Reply:
Added decompression tests for sync scenarios.